OutOfSchool/OutOfSchool.WebApi/Common/Exceptions/WorkshopNotFoundException.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace OutOfSchool.WebApi.Common.Exceptions
+{
+    [Serializable]
+    public class WorkshopNotFoundException : Exception
+    {
```

Comment:
do we really need a separate exception for each provider, teacher, workshop etc?)
what is create something like EntityNotFoundException or smth like this?))